ldms/python/ldms.pyx

```diff
@@ -3056,6 +3056,45 @@ cdef class Xprt(object):
             rc = self._conn_rc
             raise ConnectionError(rc, "Connect error: {}".format(ERRNO_SYM(rc)))
 
+    def tconnect(self, host, port=411, timeout=5):
```

Comment:
This function works very similarly to `def connect()`. I don't mind if you add `timeout` parameter to the old `connect()` function and make it timed wait instead of waiting indefinitely. I also don't mind if it is implemented separately as `tconnect()`. I'll leave the decision to combine or to separate to you :)